bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -492,12 +506,21 @@ private void resizeAndStreamLayer(SlingHttpServletResponse response, Layer layer
      *
      * @param rendition - the rendition
      * @return a layer for the rendition
-     * @throws IOException if a {@link Layer} cannot be created for the given rendition
+     *
+     * @return {@code null} if the rendition is not supported
      */
-    @NotNull
-    private Layer getLayer(@NotNull EnhancedRendition rendition) throws IOException {
+    @Nullable
```

Comment:
Added `null` return to avoid throwing an NPE when `assetHandler.getImage` was returning null (for image types that cannot be processed).